### PR TITLE
Allow emails to work if the user has URY email without server account.

### PR DESCRIPTION
**Pull request by mstratford**
Because having a user like computing@ury.org.uk does not allow myradio to email them.

### CODE REVIEW

**Comment by LordAro** [comment]
The real (other?) issue here is that someone has managed to sign up for a myradio account using a distribution list email. The sign-up should have errored and been disallowed. Then, this particular edge case disappears

**Comment by mstratford** [comment]
@LordAro Well, I guess it's our alumni form that allowed this, since it doesn't stipulate a @york.ac.uk. I was trying to test a new "welcome to URY" email format.

File: src/Classes/ServiceAPI/MyRadio_User.php [hunk 1]
```diff
@@ -499,7 +499,13 @@ public function getEmail()
                 //ffs, some people don't have an eduroam either.
                 $eduroam = $this->getEduroam();
                 if (empty($eduroam)) {
-                    return;
+                    if (!empty($this->email)) {
+                        //This is an email not associated to a user.
+                        return $this->email;
+                    } else {
+                        //Give up trying to send.
+                        return;
```
Comment (markspolakovs) [inline src/Classes/ServiceAPI/MyRadio_User.php]: This seems like a footgun to me - are we sure that all code that calls `$user->getEmail()` can handle the empty case? AFAICT no other branch returns nothing so I'm slightly concerned here

Comment (mstratford) [reply src/Classes/ServiceAPI/MyRadio_User.php]: Yup, this is how I found this bug in the first place. It tries to send an email with <> as the email address, so just fails and sticks in the MyRadio email queue.

Comment (markspolakovs) [reply src/Classes/ServiceAPI/MyRadio_User.php]: Are there any cases in which the new code could cause that failure mode as well?
